docs/reference/numbers.tex

```diff
@@ -0,0 +1,607 @@
+%!TEX root = std.tex
+\Sec0[nums]{Numbers library}
+
+\Sec1[nums.summary]{Summary}
+
+\pnum
+This Clause describes components for dealing with numbers,
+as summarized in \tref{nums.summary}.
+
+\begin{libsumtab}{Numbers library summary}{nums.summary}
+\ref{num.traits}        & Traits                                 & \tcode{<mp-units/numbers.h>} \\
+\ref{num.concepts}      & Concepts                               & \\
+\end{libsumtab}
+
+\Sec1[nums.syn]{Header \tcode{<mp-units/numbers.h>} synopsis}
+
+\indexheader{mp-units/numbers.h}%
+\begin{codeblock}
+namespace mp_units {
+
+// \ref{num.traits}, traits
+
+// \ref{num.opt.ins}, \cname{number} opt-ins
+template<typename T>
+struct enable_number;
+template<typename T>
+struct enable_complex_number;
+
+template<typename T>
+constexpr bool @\libglobal{enable_number_v}@ = enable_number<T>::value;
+template<typename T>
+constexpr bool @\libglobal{enable_complex_number_v}@ = enable_complex_number<T>::value;
+
+// \ref{num.ids}, identities
+template<typename T>
+struct number_zero;
+template<typename T>
+struct number_one;
+
+template<typename T>
+constexpr T @\libglobal{number_zero_v}@ = number_zero<T>::value;
+template<typename T>
+constexpr T @\libglobal{number_one_v}@ = number_one<T>::value;
+
+// \ref{num.assoc.types}, associated types
+template<typename T>
+struct vector_scalar;
+
+template<typename T>
+using number_difference_t = decltype(std::declval<const T>() - std::declval<const T>());
+template<typename T>
+using @\libglobal{vector_scalar_t}@ = vector_scalar<T>::type;
+
+// \ref{num.concepts}, concepts
+template<typename T>
+concept number = @\seebelow@;
+template<typename T, typename U>
+concept common_number_with = @\seebelow@;
+template<typename T>
+concept ordered_number = @\seebelow@;
+template<typename T>
+concept number_line = @\seebelow@;
+template<typename T, typename U>
+concept modulo_with = @\seebelow@;
+template<typename T, typename U>
+concept compound_modulo_with = @\seebelow@;
+template<typename T, typename U>
+concept modulus_for = @\seebelow@;
+template<typename T, typename U>
+concept compound_modulus_for = @\seebelow@;
+template<typename T>
+concept negative = @\seebelow@;
+template<typename T>
+concept set_with_inverse = @\seebelow@;
+template<typename T, typename U>
+concept point_space_for = @\seebelow@;
+template<typename T, typename U>
+concept compound_point_space_for = @\seebelow@;
+template<typename T>
+concept point_space = @\seebelow@;
+template<typename T, typename U>
+concept vector_space_for = @\seebelow@;
+template<typename T, typename U>
+concept compound_vector_space_for = @\seebelow@;
+template<typename T, typename U>
+concept scalar_for = @\seebelow@;
+template<typename T, typename U>
+concept field_for = @\seebelow@;
+template<typename T, typename U>
+concept compound_scalar_for = @\seebelow@;
+template<typename T, typename U>
+concept compound_field_for = @\seebelow@;
+template<typename T>
+concept vector_space = @\seebelow@;
```

Comment:
Did you consider also providing the `affine_space`, which should be the generalization of the `vector_space`?
See https://en.wikipedia.org/wiki/Affine_space for more details.

Reply:
I have something like `point_space_for<point_type, vector_type>`.